codeflash/code_utils/codeflash_wrap_decorator.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+import gc
+import inspect
+import os
+import pickle
+import time
+from functools import wraps
+from pathlib import Path
+from typing import Any, Callable, TypeVar
+
+from codeflash.code_utils.code_utils import get_run_tmp_file
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+def extract_test_context_from_frame() -> tuple[str, str | None, str]:
+    frame = inspect.currentframe()
+    try:
+        while frame:
+            frame = frame.f_back
+            if frame and frame.f_code.co_name.startswith("test_"):
+                test_name = frame.f_code.co_name
```

Comment:
look at how we handle such elements in our tracer and also the alvin's version of codeflash_wrap